lib/publishing.js

```diff
@@ -87,17 +87,12 @@ function createDefaultPublisher() {
     },
     GENERIC (genericNode, concretePublish) {
       const concretePublishedObjects = genericNode.objects.map(concretePublish);
-      if (genericNode.meta) {
```

Comment:
how do you know it is safe to remove the checks in this file and traversing.js? is the test coverage good enough?

Reply:
Besides the test coverage, for publishing I looked at all instances in the parser, e.g., in this case, `GENERIC`, to see whether the corresponding item should always be present (and truthy, etc.) and they were.

In the case of traversing, the recent issue of `node` being `null` had been a bug, so the guard should only have been needed when that bug was present. The definitions (and traversal implementations) of `NODE` and `NODE_LIST`, as distinct from `NULLABLE_NODE` (which already has a traversal guard against traversing nullish nodes), expect an actual node, so if there are further instances of `null` being possible, then we have a bug in our definitions anyways.

If we can get full coverage out of the parser file, which is tricky without mapping to the source file as per https://github.com/jsdoctypeparser/jsdoctypeparser/pull/105#issue-348333373 , then we can eliminate this uncertainty fully, but my thought was that any errors this might surface would be worthwhile in pointing us to fixing the source of any problem.